src/apps/common/submodules/callerId/callerId.js

```diff
@@ -93,28 +154,62 @@ define(function(require) {
 			});
 		},
 
-		callerIdUpdateNumber: function(phoneNumber, accountId, data, success, error) {
+		/**
+		 * Update a phone number
+		 * @param  {Object}   args
+		 * @param  {Object}   args.data
+		 * @param  {String}   args.data.accountId    Account ID
+		 * @param  {String}   args.data.phoneNumber  Phone number ID
+		 * @param  {Object}   args.data.data         Phone number data to be updated
+		 * @param  {Function} [args.success]         Success callback
+		 * @param  {Function} [args.error]           Error callback
+		 */
+		callerIdUpdateNumber: function(args) {
 			var self = this;
 
-			// The back-end doesn't let us set features anymore, they return the field based on the key set on that document.
-			delete data.features;
+			args.data.phoneNumber = encodeURIComponent(args.data.phoneNumber);
+
+			// The back-end doesn't let us set features anymore, they return
+			// the field based on the key set on that document.
+			delete args.data.data.features;
 
 			self.callApi({
 				resource: 'numbers.update',
-				data: {
-					accountId: accountId,
-					phoneNumber: encodeURIComponent(phoneNumber),
-					data: data
+				data: args.data,
+				success: function(data) {
+					args.hasOwnProperty('success') && args.success(data);
 				},
-				success: function(_data, status) {
-					if (typeof success === 'function') {
-						success(_data);
+				error: function(parsedError) {
+					if (parsedError.error === '402') {
```

Comment:
Now that I see this, I realize it is not ideal to have to guard against `402` errors for potentially each request handler.

Maybe we should handle this in the SDK directly so that the `error` callback is not called on such HTTP errors, I'll create a ticket for this!

Reply:
Sounds good, maybe with an option similar to the `generateError` flag? (to allow the charges event to be handled by the SDK or the error callback).